### PR TITLE
feat: download location

### PR DESCRIPTION
**Pull request by tlebon**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?
Screenshot 2024-01-18 at 18 14 55

### Issues

Updates the desktop app to allow an enforced download location

### CODE REVIEW

File: electron/src/runtime/lifecycle.ts [hunk 1]
```diff
@@ -103,15 +103,6 @@ export const quit = async (clearCache = true): Promise<void> => {
...
-    /*
-     * on MacOS, it is not possible to relaunch the app, so just fallback
-     * to reloading all the webviews
-     * see: https://github.com/electron/electron/issues/13696
-     */
-    relaunchListeners.forEach(listener => listener());
-  } else {
-    app.relaunch();
-    await quit();
-  }
```
Comment (tlebon) [inline electron/src/runtime/lifecycle.ts:116]: this hack appears to be unnecessary from my testing

**Comment by tlebon** [comment]
sorry for the translations being included, the ts wasnt allowing me to run the app once i added one translation.

File: electron/src/menu/system.ts [hunk 2]
```diff
@@ -381,16 +381,13 @@ const processMenu = (template: Iterable<MenuItemConstructorOptions>, language: l
...
-      EnvironmentUtil.platform.IS_MAC_OS ? locale.getText('menuQuit') : locale.getText('restartNow'),
-    ],
+    buttons: [locale.getText('restartLater'), locale.getText('restartNow')],
     message: locale.getText('restartLocale'),
     title: locale.getText('restartNeeded'),
     type: 'info',
   });
   if (response === 1) {
-    await (EnvironmentUtil.platform.IS_MAC_OS ? lifecycle.quit() : lifecycle.relaunch());
+    await lifecycle.relaunch();
```
Comment (atomrc) [inline electron/src/menu/system.ts:390]: is this change on purpose? It seems unrelated and sounds like a regression

Comment (tlebon) [reply electron/src/menu/system.ts:390]: Related to my other macOS change (see my first comment). The issue you highlighted previously with restart was not visible in my testing. I removed it in two locations.
